funnel/models/event.py

```diff
@@ -48,7 +49,24 @@ class Event(BaseScopedNameMixin, db.Model):
     parent = db.synonym('proposal_space')
     ticket_types = db.relationship('TicketType', secondary=event_ticket_type)
     participants = db.relationship('Participant', secondary='attendee', backref='events', lazy='dynamic')
-    __table_args__ = (db.UniqueConstraint('proposal_space_id', 'name'),)
+    __table_args__ = (db.UniqueConstraint('proposal_space_id', 'name'), db.UniqueConstraint('proposal_space_id', 'title'))
+
+    @classmethod
+    def get(cls, space, name_or_title):
+        return cls.query.filter_by(proposal_space=space)\
+            .filter(or_(cls.name == name_or_title, cls.title == name_or_title)).one_or_none()
+
+    @classmethod
+    def upsert(cls, space, name_or_title, **fields):
```

Comment:
Separate `name` and `title` here as well, making both parameters have default value `None`. In normal use your `name` is auto-generated, so you can call without a `name` parameter.